mediatimestamp/immutable.py

```diff
@@ -697,41 +717,85 @@ def to_smpte_timelabel(self, rate_num, rate_den=1, utc_offset=None):
+
+    def __mul__(self, anint: int) -> "Timestamp":
+        return cast(Timestamp, super().__mul__(anint))
 
-class TimeRange (BaseTimeRange):
+    def __rmul__(self, anint: int) -> "Timestamp":
+        return cast(Timestamp, super().__rmul__(anint))
+
+
+class TimeRange (object):
     """A nanosecond immutable precision time range object"""
 
-    EXCLUSIVE = 0x0
-    INCLUDE_START = 0x1
-    INCLUDE_END = 0x2
-    INCLUSIVE = 0x3
+    class Inclusivity (int):
+        def __and__(self, other: int) -> "TimeRange.Inclusivity":
+            return TimeRange.Inclusivity(int(self) & int(other) & 0x3)
+
+        def __or__(self, other: int) -> "TimeRange.Inclusivity":
+            return TimeRange.Inclusivity((int(self) | int(other)) & 0x3)
+
+        def __xor__(self, other: int) -> "TimeRange.Inclusivity":
+            return TimeRange.Inclusivity((int(self) ^ int(other)) & 0x3)
+
+        def __invert__(self) -> "TimeRange.Inclusivity":
+            return TimeRange.Inclusivity((~int(self)) & 0x3)
+
+    EXCLUSIVE = Inclusivity(0x0)
+    INCLUDE_START = Inclusivity(0x1)
+    INCLUDE_END = Inclusivity(0x2)
+    INCLUSIVE = Inclusivity(0x3)
 
-    ROUND_DOWN = 0
-    ROUND_NEAREST = 1
-    ROUND_UP = 2
-    ROUND_IN = 3
-    ROUND_OUT = 4
-    ROUND_START = 5
-    ROUND_END = 6
+    class Rounding(int):
+        pass
 
-    def __init__(self, start, end, inclusivity=INCLUSIVE):
+    ROUND_DOWN = Rounding(0)
+    ROUND_NEAREST = Rounding(1)
+    ROUND_UP = Rounding(2)
+    ROUND_IN = Rounding(3)
+    ROUND_OUT = Rounding(4)
+    ROUND_START = Rounding(5)
+    ROUND_END = Rounding(6)
+
+    def __init__(self,
+                 start: Optional[Timestamp],
+                 end: Optional[Timestamp],
+                 inclusivity: "TimeRange.Inclusivity" = INCLUSIVE):
         """Construct a time range starting at start and ending at end
 
         :param start: A Timestamp or None
         :param end: A Timestamp or None
         :param inclusivity: a combination of flags INCLUDE_START and INCLUDE_END"""
-        super(TimeRange, self).__init__(start, end, inclusivity)
+        super().__init__()
+        self.start: Optional[Timestamp]
+        self.end: Optional[Timestamp]
+        self.inclusivity: TimeRange.Inclusivity
+
+        self.__dict__['start'] = start
+        self.__dict__['end'] = end
+        self.__dict__['inclusivity'] = inclusivity
 
-    def __setattr__(self, name, value):
+    def __setattr__(self, name: str, value: object) -> None:
         raise TsValueError("Cannot assign to an immutable TimeRange")
 
-    def __iter__(self):
+    def __iter__(self) -> Iterator[Timestamp]:
         return self.at_rate(MAX_NANOSEC)
 
-    def __reversed__(self):
+    def __reversed__(self) -> Iterator[Timestamp]:
         return self.reversed_at_rate(MAX_NANOSEC)
 
-    def at_rate(self, numerator, denominator=1, phase_offset=TimeOffset()):
+    def at_rate(self,
+                numerator: int,
```

Comment:
The docstring says this can also be a Fraction (although I'm not sure it actually can?) Same in `reversed_at_rate`

Reply:
Right will fix. Best fix is probably to make it work if it's a fraction.